bin/autocomplete

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env ruby
+
+class AutoComplete
+  def initialize(input = ARGV[0])
+    h = eval(input)
```

Comment:
Do we need `eval` here? I'm not quite sure how this works, would you be able to explain what you're trying to do?

Also this appears to not work for several cases I've tried, e.g.

- `metal "foo.bar<tab>` gives
```
[root@controller(dev) metalware]# metal "foo.barbin/autocomplete:5:in `eval': (eval):1: syntax error, unexpected tIDENTIFIER, expecting '}' (SyntaxError)
"metal", complete: "metal "foo.bar"}
                              ^
(eval):1: unterminated string meets end of file
        from bin/autocomplete:5:in `initialize'
        from bin/autocomplete:44:in `new'
        from bin/autocomplete:44:in `<main>'
```

- in `/tmp/metalware`, `bin/metal <tab>` gives

```
[root@controller(dev) metalware]# bin/metal bin/autocomplete:23:in `add_commands': undefined method `join' for nil:NilClass (NoMethodError)
        from bin/autocomplete:13:in `run'
        from bin/autocomplete:44:in `<main>'
```